app/ide-desktop/lib/client/src/bin/server.ts

```diff
@@ -176,22 +198,62 @@ export class Server {
                 // When accessing the app from Electron, the file input event will have the
                 // full system path.
                 case '/api/upload-project': {
-                    void this.config.externalFunctions.uploadProjectBundle(request).then(id => {
+                    void this.config.externalFunctions.uploadProjectBundle(request).then(
+                        id => {
+                            response
+                                .writeHead(HTTP_STATUS_OK, [
+                                    ['Content-Length', `${id.length}`],
+                                    ['Content-Type', 'text/plain'],
+                                    ...common.COOP_COEP_CORP_HEADERS,
+                                ])
+                                .end(id)
+                        },
+                        () => {
+                            response
+                                .writeHead(HTTP_STATUS_BAD_REQUEST, common.COOP_COEP_CORP_HEADERS)
+                                .end()
```

Comment:
Should we include some information on what failed here?

Reply:
i don't think so - i don't think we have more specific information, so the default "<action x> failed" message that the frontend outputs for most errors should be sufficient